src/main/java/be/seeseemelk/mockbukkit/enchantments/EnchantmentMock.java

```diff
@@ -193,14 +254,14 @@ public boolean isCursed()
 	 */
 	public void setCursed(boolean isCursed)
 	{
-		this.isCursed = isCursed;
+		this.cursed = isCursed;
 	}
 
 	@Override
 	public boolean conflictsWith(@NotNull Enchantment other)
 	{
-		Preconditions.checkNotNull(other, "Enchantment cannot be null");
-		return false;
+		Preconditions.checkArgument(other != null, "others can't be null") ;
```

Comment:
Why was `checkNotNull` replaced `checkArgument`? We shouldn't copy Spigot's bad practices. If anything, we should convince upstream to use the proper method.

Reply:
Because we are checking an Argument. Usually checkNotNull should be used later on to check if something is not null, e.g. after trying to retrieve an object from a collection

Reply:
From the checkNotNull Javadoc:
> Ensures that an object reference passed as a parameter to the calling method is not null.

Reply:
I still think that checkArgument is the better choice here

Reply:
Why? We're checking to see if it's null, and there's an explicit method & exception for precisely that purpose.

Reply:
If we take the actual code that we are mocking into relation, it looks like this:
```java
    @Override
    public boolean conflictsWith(Enchantment other) {
        if (other instanceof EnchantmentWrapper) {
            other = ((EnchantmentWrapper) other).getEnchantment();
        }
        if (!(other instanceof CraftEnchantment)) {
            return false;
        }
        CraftEnchantment ench = (CraftEnchantment) other;
        return !this.handle.isCompatibleWith(ench.getHandle());
    }
```
It will in essence just return false.